spring-batch-core/src/main/java/org/springframework/batch/core/step/AbstractStep.java

```diff
@@ -287,14 +294,8 @@ public final void execute(StepExecution stepExecution)
 						name, stepExecution.getJobExecution().getJobInstance().getJobName()), e);
 			}
 			stopObservation(stepExecution, observation);
```

Comment:
I think this should also be moved before the listener. Otherwise there will be a difference between the calculated stepExecutionDuration and the micrometer metric. I will take care of that change on merge.

Reply:
I tried this but it breaks the step status in the metric. I will leave it as is.